Yank/restraints.py

```diff
@@ -313,8 +320,31 @@ class ReceptorLigandRestraint(ABC):
         4. Optionally, implement :func:`determine_missing_parameters` to fill in
         the parameters left undefined in the constructor.
 
+        5. Optionally allow the setting of the ``restraint_name`` keyword arg
+
     """
 
+    def __init__(self, restraint_name=None):
+        self._restraint_name = restraint_name
+
+    @property
+    def restraint_name(self):
+        """Provide the restraint_name suffix"""
+        return self._restraint_name
+
+    @property
+    def full_restraint_name(self):
```

Comment:
I would call this something like `controlling_parameter_name` to make the connection clear to the `openmmtools` stuff and because technically I wouldn't say that `lambda_restraints_myrestraintname` is a name for the restraint.